src/app/services/auth/AuthService.js

```diff
@@ -0,0 +1,42 @@
+const config = require('config')
+const bcrypt = require('bcrypt')
+const jwt = require('jsonwebtoken')
+
+const log = require('../../util/logger')
+const serviceLocator = require('../serviceLocator')
+
+const SECRET = 'sssshhhhhhhh'
+
+class AuthService {
+  constructor() {
+    this._userService = serviceLocator.userService()
+  }
+
+  /**
+   * Register a new user and return jwt
+   * @param {RegisterDTO} registerDto
+   * @return {Promise<string>} - The jwt token
+   */
+  async register(registerDto) {
+    const hashedPassword = bcrypt.hashSync(registerDto.password, 8)
+
+    const user = await this._userService.create({
+      email: registerDto.email,
+      hashedPassword
+    })
+
+    const secret = 'secret'
+    const token = jwt.sign({id: user.id}, SECRET, {
+      expiresIn: '5m'
+    })
```

Comment:
Pulled all this from: https://medium.freecodecamp.org/securing-node-js-restful-apis-with-json-web-tokens-9f811a92bb52